src/core/include/openvino/core/model.hpp

```diff
@@ -24,9 +24,18 @@
 #include "openvino/runtime/tensor.hpp"
 
 namespace ov {
+class Model;
+void copy_shared_objects(ov::Model& dst, const ov::Model& src);
+namespace frontend {
+class FrontEnd;
+}
 class ModelAccessor;
 /// A user-defined function.
 class OPENVINO_API Model : public std::enable_shared_from_this<Model> {
+    friend class ov::frontend::FrontEnd;
+    friend void copy_shared_objects(ov::Model& dst, const ov::Model& src);
```

Comment:
BTW, we already have `ModelAccessor` in this file, can it be reused?

Reply:
I'm not sure, in this case any other objects may have an access to 'm_shared_object' via this ModelAccessor. Not sure that we want it